### PR TITLE
[RateLimiter] Fix sleep value

### LINKED ISSUE #40656: [Bug][RateLimiter] sleep/usleep typo
**Symfony version(s) affected**: 5.x

**Description**
https://github.com/symfony/rate-limiter/blob/5.x/RateLimit.php#L70

Probably a typo and there should be `usleep`? (or remove `1e6` )

```
    public function wait(): void
    {
        sleep(($this->retryAfter->getTimestamp() - time()) * 1e6);
    }
```

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | 5.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40656
| License       | MIT
| Doc PR        | 

sleep vs usleep

### CODE REVIEW

File: src/Symfony/Component/RateLimiter/RateLimit.php [hunk 1]
```diff
@@ -67,6 +67,6 @@ public function getLimit(): int
 
     public function wait(): void
     {
-        sleep(($this->retryAfter->getTimestamp() - time()) * 1e6);
+        usleep(($this->retryAfter->getTimestamp() - time()) * 1e6);
```
Comment (Nyholm) [inline src/Symfony/Component/RateLimiter/RateLimit.php]: Let's make this simpler: 

```suggestion
        sleep($this->retryAfter->getTimestamp() - time());
```

Alternatively, we decide that we want to support microseconds and use `usleep` and `microtime`

**Review by Nyholm — CHANGES_REQUESTED** [review]
Your change is correct. But I have one suggestion.

File: src/Symfony/Component/RateLimiter/RateLimit.php (hunk 1 above)
Comment (jderusse) [reply src/Symfony/Component/RateLimiter/RateLimit.php]: indeed, time() has `1 sec` precision.

Thanks

**Review by Nyholm — APPROVED** [review]
Thank you

**Comment by Nyholm** [comment]
Btw, doing `sleep(-3)` is a fatal error in PHP8. (https://3v4l.org/lK3uR)

Are we sure this method is always called when there are time left to wait? Or should we do `max(0, ...)` just to be sure?

**Review by Nyholm — APPROVED** [review]
Thank you. 

Sorry for all the comments on this small patch.

**Comment by nicolas-grekas** [comment]
Thank you @jderusse.

**Pull request merged**